### PR TITLE
Animate the typing notification bar in and out of the RoomScreen view

### LINKED ISSUE #155: Improve the `typing_notice` view with animations
For the context of this issue, please refer to #99  and [PR 147](https://github.com/project-robius/robrix/pull/147#issuecomment-2365230896)  . 

The functions to be implemented under this issue are:

- [x] Make the display interface of the ongoing notification more readable.
- [x] Add an animated ellipsis after the "Users are typing" label
- [ ] animate the `typing_notice` view in and out of the room screen view, such as by having it fold up or fade upwards from the bottom

### PR DESCRIPTION
**Pull request by alanpoon**
https://github.com/user-attachments/assets/f1e2f667-957d-4a8b-9625-c4b9cd0de007

Fixed 3rd task of this issue
https://github.com/project-robius/robrix/issues/155

Closes #155

### CODE REVIEW

**Comment by ZhangHanDong** [comment]
@alanpoon 
The vertical height of the notification background is a bit too large, can it be adjusted to be narrower? 

Also, resolve the conflict.

**Comment by alanpoon** [comment]
> @alanpoon The vertical height of the notification background is a bit too large, can it be adjusted to be narrower?
> 
> Also, resolve the conflict.

Done.

**Review by kevinaboos — CHANGES_REQUESTED** [review]
Nice job, looks really good! 

Can you also make it sliding upwards when showing it too? That way the animation is symmetric when being shown and when being hidden.

I also think it'd look nicer if the animation was a little bit faster (like 25% faster), but that's ultimately up to you.

**Comment by alanpoon** [comment]
> Nice job, looks really good!
> 
> Can you also make it sliding upwards when showing it too? That way the animation is symmetric when being shown and when being hidden.
> 
> I also think it'd look nicer if the animation was a little bit faster (like 25% faster), but that's ultimately up to you.

- Added Sliding upwards when showing. 
- Sped up the animation
- Height of typing_notice made narrower

https://github.com/user-attachments/assets/8dbc8eb2-3e45-4c41-a2ab-7a26cc09f94c

**Review by kevinaboos — DISMISSED** [review]
Looks great, thanks very much!

Please resolve conflicts and then I will merge it in.

**Comment by alanpoon** [comment]
resolved conflict and merged main

**Review by kevinaboos — APPROVED** [review]

File: src/home/room_screen.rs [hunk 1]
```diff
@@ -1595,15 +1616,11 @@ impl RoomScreen {
...
                         }
                     };
-                    let is_typing = !users.is_empty();
-                    self.view.view(id!(typing_notice)).set_visible(is_typing);
-                    self.view.label(id!(typing_label)).set_text(&typing_text);
-                    let typing_animation = self.view.typing_animation(id!(typing_animation));
-                    if is_typing {
-                        typing_animation.animate(cx);
-                    } else {
-                        typing_animation.stop_animation();
```
Comment (alanpoon) [inline src/home/room_screen.rs:1605]: [typing animat](https://github.com/project-robius/robrix/issues/259)

**Pull request merged**